lib/livebook/hubs/team_client.ex

```diff
@@ -397,6 +406,34 @@ defmodule Livebook.Hubs.TeamClient do
     |> dispatch_deployment_groups(agent_connected)
   end
 
+  defp handle_event(:agent_key_created, agent_key_created, state) do
+    agent_key = build_agent_key(agent_key_created)
+
+    case find_deployment_group(agent_key.deployment_group_id, state.deployment_groups) do
+      nil ->
+        state
+
+      deployment_group ->
+        agent_keys = Enum.uniq([agent_key | deployment_group.agent_keys])
```

Comment:
Is there a case where we get a new key and it's a duplicate?

Reply:
It's just a precaution because of the synchronous processing of messages via the API response and receiving the event via WebSocket, so as not to duplicate for some reason

Reply:
I might overthinking there, but your call.

Reply:
Ah, so in that case whoever creates the key always gets two events, right? I see for secret we do remove + add. We could also look for agent key by id and if present just ignore the event.

Reply:
Only the creator gets two events, the other users connected to WebSocket would receive only one

Reply:
Yeah, it would work too